sklearn/preprocessing/data.py

```diff
@@ -1182,6 +1182,35 @@ def powers_(self):
         return np.vstack(np.bincount(c, minlength=self.n_input_features_)
                          for c in combinations)
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "x_n_features" is used.
```

Comment:
Maybe it would be cleaner without underscores? I find things like ``x_1^2 x_2^1`` harder to visually parse than ``x1^2 x2^1``